CMakeLists.txt

```diff
@@ -287,6 +287,25 @@ foreach(EXECUTABLE ${IRODS_CLIENT_ICOMMANDS_EXECUTABLES})
   endif()
 endforeach()
 
+# Fix build on EL7
+if (IRODS_LINUX_DISTRIBUTION_NAME STREQUAL "centos" AND IRODS_LINUX_DISTRIBUTION_VERSION_MAJOR STREQUAL "7" AND NOT IRODS_BUILD_AGAINST_LIBCXX)
+  set(
+    ICOMMANDS_NEEDING_OVERLINK
```

Comment:
What does this mean?

I want to make sure my mental model of `NEEDING_OVERLINK` is correct.

Reply:
it's just the name of the variable I'm using to store the list of icommands that need to be linked with `--no-as-needed`

Reply:
Sorry, I'm referring to the use of `OVERLINK`. What does that mean to you? I'm not familiar with that term.

Reply:
Oh, it's not a standard term or anything. `--no-as-needed` tells the linker not to drop link libraries it deems unnecessary, so when it's used, it can result in the binary being linked to more libraries than it would have otherwise.

Reply:
I see. Thanks for the explanation.